zml/ops.zig

```diff
@@ -138,7 +137,7 @@ pub fn reduce(
     var init_values: [N]mlir.Value = undefined;
     ctx.extractValues(&inits, &init_values);
 
-    const body_block, _ = ctx.makeBlock(BodyS, &body_fn, {}, .{ inits, inits });
+    const body_block, _ = ctx.makeBlock(.hermetic, BodyS, &body_fn, {}, .{ inits, inits });
```

Comment:
Just curious, `reduce` operations are not allowed to access outer scope ?

Reply:
I thought so, but I tested generating some IR that was using a constant from above and it worked. But since `zml.ops.reduce` doesn't take a context, it's not possible in ZML.
Maybe we should add a `reduceWithCtx` ?

Reply:
at least stablehlo-opt doesn't complain, but I haven't tried actually running anything like that. Maybe leave it like that until we have a usecase ?

Reply:
One usecase I see would be a `scan` like function that could prove useful for flash attention
But yeah, in the meantime we can keep it like this